### DISCUSSION TITLE
How to sign appxbundle and appx using signtool with usb token?

### DISCUSSION THREAD

**Discussion by HppZ** [microsoft/WindowsAppSDK#1081] [Q&A] [answered]
I have a question and I really need your help:

When sign a appxbundle or appx file, signtool gives me an error:
Multiple signature supports is not implemented for this filetype.

Command line is: 
signtool sign /tr http://sha256timestamp.ws.symantec.com/sha256/timestamp /a /as /fd SHA256 FILE.appxbundle

or 

signtool sign /v /as /fd sha256 /sha1 THUMBPRINT_HERE /tr http://rfc3161timestamp.globalsign.com/advanced /td sha256 FILE.appxbundle

it works fine if I replace FILE.appxbundle with FILE.exe.

so does signtool support appxbundle and appx file types?
Do I have to install windows SDK?

signtool version: 10.0.19041.685
windows 19042

PLEASE HELP ME.

**Comment by riverar** [comment, marked as answer]
@HppZ Signtool supports signing appx/msix. As the error suggests, you are trying to sign a file that's already signed. Can you verify that?

**Reply by HppZ** [reply]
I tried pfx file, it works. but I only have usb token from GlobalSign, because then don't provide pfx file.

**Reply by HppZ** [reply]
publisher name can't contain Chinese char?

**Reply by riverar** [reply]
Interesting, looks like a bug as `UTF8String` should be acceptable. Adding @jvintzel @aclinick to take a look.

**Reply by HppZ** [reply]
https://github.com/dotnet/arcade/issues/825
> The certificate that we are using to sign the file is trying to sign the file twice and that is the reason of the error.

**Reply by HppZ** [reply]
problem solved, just remove /as option.
